src/iterative_sorting/iterative_sorting.py

```diff
@@ -2,18 +2,26 @@
 
 
 def selection_sort(arr):
-    for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
+    for i in range(0, len(arr) - 1): #Begin from min to high index 
+        cur_index = i #Holds min position, changing the min position to i every after iteration | starts with zero
+        #smallest_index = cur_index
         # TO-DO: find next smallest element
         # (hint, can do in 3 loc)   
-        for j in range(cur_index, len(arr)):
-            if arr[smallest_index] > arr[j]:
-                smallest_index = j
+        
+        #every after iteration we are creating a sorted array so we have to reduce the size of the array
+        for j in range(cur_index, len(arr)): #size of the array/it will go up to the inner loop
+            if arr[j] < arr[cur_index]: #change the min position
+                cur_index = j #once you find a value, change the min position : now it has new position
     
 
         # TO-DO: swap
```

Comment:
Nice nice job Itel! Sorting is one of those interesting sorts in that it reaches all the way as far as it can into the rest of the unsorted away, and goes all the way back once it finds that min value.